deps/build.jl

```diff
@@ -1,59 +1,36 @@
-include("../src/compat.jl")
-
-# version of library to download
-const version = v"3.2.1"
-
-using BinDeps
-@BinDeps.setup
-
-if !isdefined(Base, :Libdl)
-	# TODO for Julia upgrade: remove check
-	using Libdl
-end
-
-function compatible_version(lib, handle)
-	major, minor, rev = Ref{Cint}(), Ref{Cint}(), Ref{Cint}()
-	ccall(Libdl.dlsym(handle, :glfwGetVersion), Cvoid, (Ref{Cint}, Ref{Cint}, Ref{Cint}), major, minor, rev)
-	libversion = VersionNumber(major[], minor[], rev[])
-	return libversion >= version
-end
-
-glfw = library_dependency("libglfw", aliases=["glfw", "glfw3", "libglfw3"], validate=compatible_version)
-
-# library source code
-provides(Sources, URI("https://github.com/glfw/glfw/archive/$version.tar.gz"), glfw, unpacked_dir="glfw-$version")
-srcdir = joinpath(BinDeps.srcdir(glfw), "glfw-$version")
-
-# how to build library from source
-cmake_options = map(x -> "-D$(x[1])=$(x[2])", [
-	("BUILD_SHARED_LIBS",    "ON"),
-	("CMAKE_INSTALL_PREFIX", BinDeps.usrdir(glfw)),
-	("GLFW_BUILD_DOCS",      "OFF"),
-	("GLFW_BUILD_EXAMPLES",  "OFF"),
-	("GLFW_BUILD_TESTS",     "OFF")
-])
-cmake_build_steps = @build_steps begin
-	GetSources(glfw)
-	@build_steps begin
-		ChangeDirectory(srcdir)
-		`cmake $cmake_options .`
-		MakeTargets("install")
-	end
-end
-provides(SimpleBuild, cmake_build_steps, glfw)
-
-# get library through Homebrew, if available
-if isapple()
-	using Homebrew
-	provides(Homebrew.HB, "glfw", glfw, os=:Darwin)
-end
-
-# download a pre-compiled binary (built by GLFW)
-if iswindows()
-	archive = "glfw-$version.bin.WIN$(Sys.WORD_SIZE)"
-	libpath = joinpath(archive, "lib-mingw-w64")
-	uri = URI("https://github.com/glfw/glfw/releases/download/$version/$archive.zip")
-	provides(Binaries, uri, glfw, unpacked_dir=archive, installed_libpath=libpath, os=:Windows)
+using BinaryProvider # requires BinaryProvider 0.3.0 or later
+
+# Parse some basic command-line arguments
+const verbose = "--verbose" in ARGS
+const prefix = Prefix(get([a for a in ARGS if a != "--verbose"], 1, joinpath(@__DIR__, "usr")))
+products = [
+    LibraryProduct(prefix, "", :lib),
+]
+
+# Download binaries from hosted location
+bin_prefix = "https://github.com/SimonDanisch/GLFWBuilder/releases/download/3.2.1"
+
+# Listing of files generated by BinaryBuilder:
+download_info = Dict(
+    Windows(:i686) => ("$bin_prefix/glfw.v1.0.0.i686-w64-mingw32.tar.gz", "08bdec95b829c46f5aee3d98d4ba4a29eb79c0553f5ec68749d926598fa1f4b8"),
+    MacOS(:x86_64) => ("$bin_prefix/glfw.v1.0.0.x86_64-apple-darwin14.tar.gz", "7b4c712c0e194c7196750b14d3178ba8ece8eeeed0fbac64f3b6e90579f0efda"),
```

Comment:
What does the `14` in the `darwin14` denote? Do you know how many versions of OS X this build will work for?

Reply:
no idea^^